pkg/crc/services/dns/dns.go

```diff
@@ -128,7 +128,15 @@ func CheckCRCPublicDNSReachable(serviceConfig services.ServicePostStartConfig) (
 		}
 		curlArgs = append(curlArgs, "--noproxy", proxyConfig.GetNoProxyString())
 		if proxyConfig.ProxyCAFile != "" {
-			curlArgs = append(curlArgs, "--proxy-cacert", proxyConfig.HTTPProxy)
+			// --proxy-cacert/--cacert replaces the system CAs with the specified one.
+			// If not using MITM proxy, --cacert must *not* be used, and if not using
+			// https:// proxy, --proxy-cacert must *not* be used
+			// ProxyCAFile is ambiguous, we cannot know if it's set because of MITM proxy,
+			// because of https:// proxy, or because of both
+			// We do not really care about transport security for this test, all that
+			// matters is whether or not we can resolve the hostname, so we can
+			// workaround this ambiguity by using an insecure connection
+			curlArgs = append(curlArgs, "--insecure", "--proxy-insecure")
```

Comment:
@cfergeau I was reading `man curl` and also searched bit on internet but didn't get this info which you put as comment. Can you provide a link to read around it and add it in the comment also?

Reply:
"this info", "around it"? I'm not sure which part of the comment is not clear/needs a link? I did not use more than the man page.
Short list of steps performed by `curl quay.io`:
1. if a proxy is in use, connect to it. If this is a https proxy, this involves creating a tls connection. cert checks/... can be bypassed with --proxy-insecure
2. DNS resolution of quay.io (will be performed either locally, or by the proxy)
3. connect to quay.io. If the connection is done to a https URL, this involves creating a tls connection. cert checks/... can be bypassed with --insecure

The step we want to test is 2., TLS failures at step 1. or 3. are not particularly worrying/useful for this test.